packages/gui/src/components/nfts/gallery/NFTfilteredNFTs.tsx

```diff
@@ -0,0 +1,91 @@
+import { useMemo } from 'react';
+import { useGetNFTWallets } from '@chia/api-react';
+import useFetchNFTs from '../../../hooks/useFetchNFTs';
+import useHideObjectionableContent from '../../../hooks/useHideObjectionableContent';
+import useNachoNFTs from '../../../hooks/useNachoNFTs';
+import useNFTMetadata from '../../../hooks/useNFTMetadata';
+import type { NFTInfo } from '@chia/api';
+
+type FilteredNFTtype = {
+  walletId: number | undefined;
+};
+
+export default function useFilteredNFTs(props: FilteredNFTtype) {
+  const { walletId } = props;
+  const { wallets: nftWallets, isLoading: isLoadingWallets } =
+    useGetNFTWallets();
+  let { nfts, isLoading: isLoadingNFTs } = useFetchNFTs(
+    nftWallets.map((wallet: Wallet) => wallet.id),
```

Comment:
nftWallets can be undefined here if `useGetNFTWallets()` is still loading.

```suggestion
    nftWallets?.map((wallet: Wallet) => wallet.id) ?? [],
```